awx/api/filters.py

```diff
@@ -395,11 +395,11 @@ def filter_queryset(self, request, queryset, view):
                         order_by = value.split(',')
                     else:
                         order_by = (value,)
-            if order_by is None:
-                order_by = self.get_default_ordering(view)
+            default_order_by = self.get_default_ordering(view)
+            # glue the order by and default order by together so that the default is the backup option
+            order_by = list(order_by or []) + list(default_order_by or [])
```

Comment:
With the change to the view ordering, I'm not sure if this change is necessary.  I also don't like that it would be changing the filter class behavior in the case that `order_by` is `None`.

Reply:
This change is absolutely necessary.  The original problem is when someone has clicked on the 'status' header, the sort is unstable.  This is because the usual case is that the value is the empty string.  Our goal here is to have the parameters passed to Django's `.order_by()` be `['errors', 'id']`, so that there is a fall-back field resulting in a stable sort.

Reply:
Previously when the UI passes in _any_ parameter for `order_by`, the default ordering specified by the view gets thrown away entirely.  Now it will be appended to the end of any parameters passed in by the UI as a fall back.

Reply:
There _may_ be instances where this changes the actual ordering that a view results in, but it can only come up when a user-requested sort is used and _only_ as the final fall-back field, so I think it will be quite rare and unobtrusive.

Reply:
This is going to impact _all_ list views. I see what you're saying - the prior behavior was that when `order_by` was None it used the view's default ordering (seems unnecessary, but not something this is changing). My last comment was wrong.

But I would like to see a test or two, because this could be a relatively high-impact change. While the issue filed was for instances, I'd expect that similar bugs can be seen in other parts of the UI (order projects by scm_type, right?)

Reply:
> But I would like to see a test or two, because this could be a relatively high-impact change. While the issue filed was for instances, I'd expect that similar bugs can be seen in other parts of the UI (order projects by scm_type, right?)

Probably.

Reply:
I am currently writing some tests. If there are any particular scenarios you're concerned about let me know so I can account for them specifically.